drivers/platform/x86/asus-wmi.c

```diff
@@ -1619,6 +1625,25 @@ static int is_display_toggle(int code)
 	return 0;
 }
 
+static bool asus_wmi_is_fnlock_enabled(struct asus_wmi *asus)
+{
+	u32 result;
+
+	asus_wmi_get_devstate(asus, ASUS_WMI_DEVID_FNLOCK, &result);
+
+	if (result & ASUS_WMI_DSTS_STATUS_BIT)
```

Comment:
This is checking bit 0, which in the case of this device, is not a status bit. It's documented as "Fn + ESC is disabled in BIOS menu". So perhaps add a new define for clarity.

Also, to prevent the FnLock code running on platforms that don't have it, I think you also need to fall back to the standard ASUS_WMI_DSTS_PRESENCE_BIT	check.

You can do all this in a single return statement e.g.
`return (result & ASUS_WMI_DSTS_PRESENCE_BIT) && !(result & ASUS_WMI_FNLOCK_BIOS_DISABLED);`

Reply:
I also thought the BIT 0 = 1 should indicate it's enabled in BIOS. But I found when Fn-Lock Option disabled in BIOS, the BIT 0 of query DSTS (device status) will then be 1. A new definition of `ASUS_WMI_FNLOCK_BIOS_DISABLED` makes this function better understood. Thanks